File: packages/sdk-python/src/eidolondb/resources/ingest.py
from __future__ import annotations

from typing import Any

from .._client import EidolonDBClient
from .._types import IngestResponse, IngestSource


class IngestResource:
    def __init__(self, client: EidolonDBClient):
        self._client = client

    def run(
        self,
        content: str,
        *,
        source: IngestSource = "chat",
        auto_store: bool = True,
        debug: bool = False,
        **kwargs: Any,
    ) -> IngestResponse:
        payload = {
            "content": content,
            "source": source,
            "autoStore": auto_store,
            "debug": debug,
            **kwargs,
        }
        return self._client.request("POST", "/ingest", body=payload)

File: packages/sdk-js/src/resources/ingest.ts
import { EidolonDBClient } from '../client.js';
import type { IngestRequest, IngestResponse } from '../types.js';

export class IngestResource {
  constructor(private readonly client: EidolonDBClient) {}

  /** Run ingest extraction + dedup + optional auto-store. */
  run(input: IngestRequest): Promise<IngestResponse> {
    return this.client.request<IngestResponse>('POST', '/ingest', input);
  }
}
